src/TabBarIndicator.tsx

```diff
@@ -6,39 +6,107 @@ import memoize from './memoize';
 import { Route, SceneRendererProps, NavigationState } from './types';
 
 export type Props<T extends Route> = SceneRendererProps & {
+  dynamicWidth?: boolean;
   navigationState: NavigationState<T>;
+  tabWidths: number[];
   width: number;
   style?: StyleProp<ViewStyle>;
 };
 
 const { max, min, multiply } = Animated;
 
-export default class TabBarIndicator<T extends Route> extends React.Component<
-  Props<T>
-> {
+export default class TabBarIndicator<
+  T extends Route
+> extends React.PureComponent<Props<T>> {
   private getTranslateX = memoize(
-    (position: Animated.Node<number>, routes: Route[], width: number) =>
-      multiply(
+    (
+      position: Animated.Node<number>,
+      routes: Route[],
+      tabWidths: number[],
+      width: number,
+      dynamicWidth?: boolean
+    ) => {
+      if (dynamicWidth) {
+        let inputRange: number[] = [];
+        let outputRange: number[] = [];
+        let totalWidth = 0;
+        routes.forEach((_route: Route, i: number) => {
```

Comment:
Instead of a `forEach`, I feel a `map` for `inputRange` and a `reduce` for `outputRange` is cleaner

Reply:
done